### ISSUE TITLE
Playback fails with "No video formats found"

### ISSUE THREAD

**Issue by JaskaRoman** [natumbri/mopidy-youtube#147]
I'm able to play some videos but for others, the playback fails with "No video formats found", which appears to be ralated to regional restrictions.
The [URL](https://www.youtube.com/watch?v=e8X3ACToii0) I am trying to play works with the regular youtube-dl command on its own.
I did not set up any proxies.
Everything is up to date, running in a Hyper-V Ubuntu Server.

Is there something I am missing?

**Comment by natumbri** [comment]
@JaskaRoman Thanks for the bug report.  No, nothing you are missing that I can think of - I've had a go at playing it and it seems to work fine for me (in Australia).  Any chance you could try a VPN?

**Comment by JaskaRoman** [comment]
@natumbri Thank you for a swift reply. I'm currently unable to verify using a VPN, but from what you are saying, it does appear to be a regional issue (I am located in Czech republic). What is the difference between running a direct youtube-dl and using mopidy-youtube? Is it possible, that the problem lays in the internal track look-up or some exotic youtube-dl parameters being used?
I have briefly tried to make sense of the code, but I'm not much of a python programmer. 
As I see it, the first step is to reproduce the behavior outside of mopidy-youtube. What commands would I call in order to do this?

**Comment by natumbri** [comment]
@JaskaRoman sounds like a good plan.  The call to youtube_dl is here:

```
               info = youtube_dl.YoutubeDL(
                    {
                        "format": "bestaudio/best",
                        "proxy": self.proxy,
                        "nocheckcertificate": True,
                        "cachedir": False,
                    }
                ).extract_info(
                    url="https://www.youtube.com/watch?v=%s" % self.id,
                    download=False,
                    ie_key=None,
                    extra_info={},
                    process=True,
                    force_generic_extractor=False,
                )
```
Where `self.id` is a 11 character youtube video id (eg `9TlBTPITo1I`).  Proxy is `None`, if you haven't set one.

The headers in the request are:
```
        headers = {
            "user-agent": httpclient.format_user_agent(self.user_agent),
            "Cookie": "PREF=hl=en;",
            "Accept-Language": "en;q=0.8",
        }
```
The user agent is `Extension.dist_name/Extension.version` which translates  to `Mopidy-YouTube/3.0`.

Is that enough for you to try to replicate outside of the extension?

**Comment by JaskaRoman** [comment]
@natumbri Thank you, I'll give it a try and report back.

**Comment by kingrun** [comment]
I have probably the same problem as well. Ill post the Mopidy log below. Basically some songs do play and some dont. I use Mopidy Party for the search feature. Bu it still uses mopidy-youtube for playblack and it only plays some songs. maybe 1 or 2 out of 10. As far as I know within mopidy.conf there is a setting for proxy options but I have no idea how to set it up as if I set up a HTTP proxy that I found online it always fails to connect. Since I know regularly these videos work. And I used this setup before the upgrade to 3.0 it worked as well. I cant seem to figure out the issue.

> Jun 05 06:27:07 audiocentras mopidy[351]: ERROR    [Thread-15] mopidy_youtube audio_url error "ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt
> 
> Jun 05 06:27:07 audiocentras mopidy[351]: WARNING  [Core-7] mopidy.core.tracklist Track is not playable: youtube:video/AC Slater - Bass Inside.xHTbqAfFbVo
> Jun 05 06:27:07 audiocentras mopidy[351]: INFO     [YouTubeBackend-6] mopidy_youtube youtube PlaybackProvider.translate_uri "youtube:video/AC Slater - Bass Inside (Official Music Video).6XJkPtxqWM8"
> Jun 05 06:27:07 audiocentras mopidy[351]: ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call you
> Jun 05 06:27:07 audiocentras mopidy[351]: ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. 
> 

[journal.txt](https://github.com/natumbri/mopidy-youtube/files/4734110/journal.txt)

**Comment by natumbri** [comment]
thanks @kingrun - this is a tough one for me to debug, as they play fine for me!  Interesting that it worked pre-3.0.

**Comment by kingrun** [comment]
Yeah i have left it lying around for a few months and when I booted back up nothing worked so I figured an update is in order. And then this happened . The problem is that some do actually play while most dont, Id say maybe 2 songs out of 10 i try. If non would play that might be a concrete issue. But it seems the same as @JaskaRoman Though I cant try a VPN cause my setup is configured to route to my home vpn since I use it with my mobile data on the go at places where there is no internet. Right now my bypass for this is to use soundcloud and spotify. But I have no idea as well how to troubleshoot this. It might be a region restriction issue, but that's the point that it didn't happen before the update.

**Comment by natumbri** [comment]
Upgrading from 2 to 3 has caused problems for people as a result of old versions of various things (python modules) being left lying around ... Could that have happened to you?

**Comment by JaskaRoman** [comment]
I was not able to reproduce the problem using the direct youtube-dl call with provided agruments.
I will try to set up a clean install.

**Comment by natumbri** [comment]
Ok - let me know how you go!

**Comment by JaskaRoman** [comment]
I have finally found the time to set the whole thing up again, which proved to be more tedious than I expected.

First I tried to get it running on ubuntu server 20, which is the current LTS. After several hours I found out that the included `libshout3` version (`2.4.3` iirc) has a bug which breaks streaming. People recommended downgrading to `2.4.1`, but the version is not available in the default feed and I wanted to go as vanilla as possible.

Next, I decided to try using the previous LTS version - the ubutnu server 18, which I quickly found out is no longer compatible with mopidy, which requires 19+.

So the third time I tried, I went back to the initial OS - ubuntu server 19, which is dying next month. I set everything up using the guides at [mopidy docs](https://docs.mopidy.com/en/latest/installation/debian/) and [this repo's readme](https://github.com/natumbri/mopidy-youtube) and I was finally able to successfully play the tracks, which could not be played before.

TL;DR - User error - setting up everything from scratch fixed the issue.

**Comment by natumbri** [comment]
@JaskaRoman sorry to hear that it was such a tedious process - I also ran into issues with mopidy 3.0 and Ubuntu 18 - but glad to hear you got it working!
